code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -267,7 +267,28 @@
 	else
 		M.visible_message("<span class='notice'>[M] hugs [src] to make [p_them()] feel better!</span>", \
 					"<span class='notice'>You hug [src] to make [p_them()] feel better!</span>")
-		SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "hug", /datum/mood_event/hug)
+
+		// Warm them up with hugs
+		if(bodytemperature > M.bodytemperature) // they are warmer leech from them
+			var/temp_diff = bodytemperature - M.bodytemperature
+			M.adjust_bodytemperature(min((1 - M.get_insulation_protection(bodytemperature)) * temp_diff / BODYTEMP_HEAT_DIVISOR, \
+			BODYTEMP_HEATING_MAX))
+			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "hug", /datum/mood_event/warmhug, src)
+			SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "hug", /datum/mood_event/hug)
+
+		else // you are warm share the heat of life
+			var/temp_diff = M.bodytemperature - bodytemperature
+			adjust_bodytemperature(min((1 - get_insulation_protection(M.bodytemperature)) * temp_diff / BODYTEMP_HEAT_DIVISOR, \
+			BODYTEMP_HEATING_MAX))
+			SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "hug", /datum/mood_event/warmhug, M)
+
+		// Let people know if they hugged someone really warm
+		if(M.bodytemperature > BODYTEMP_HEAT_DAMAGE_LIMIT)
+			to_chat(src, "<span class='warning'>[M] felt like they are over heating when they hugged you.</span>")
+
+		if(bodytemperature > BODYTEMP_HEAT_DAMAGE_LIMIT)

```

Comment:
add messages for hugging/being hugged by people who are freezing too

Reply:
done